src/api/provider.ts

```diff
@@ -60,6 +61,6 @@ export class FetchProvider implements APIProvider {
         } catch {
             // ignore json parse errors
         }
-        return {status: response.status, json, text}
+        return {headers: response.headers, status: response.status, json, text}
```

Comment:
We should normalize the headers to a vanilla js object with string keys and values otherwise it will lock us in to the fetch header object which other api providers might not have.

Reply:
Alright, I think I have all of this feedback resolved. 

Check it out when you get the chance, and if it looks good, I'm done with it 👍